make/integration-tests.mk

```
@@ -0,0 +1,47 @@
+INTEGRATION_COVER_PKGS = ./pkg/...,./controllers/...,./api/...
+
+##@ Integration tests
+
+.PHONY: test-bare-k8s-integration
+test-bare-k8s-integration: clean-cov generate fmt vet ginkgo ## Requires only bare kubernetes cluster.
+	mkdir -p $(PROJECT_PATH)/coverage/bare-k8s-integration
+#	Check `ginkgo help run` for command line options. For example to filtering tests.
+	$(GINKGO) \
+		--coverpkg $(INTEGRATION_COVER_PKGS) \
+		--output-dir $(PROJECT_PATH)/coverage/bare-k8s-integration \
+		--coverprofile cover.out \
+		-tags integration \
+		./tests/bare_k8s/...
+
+.PHONY: test-gatewayapi-env-integration
+test-gatewayapi-env-integration: clean-cov generate fmt vet ginkgo ## Requires kubernetes cluster with GatewayAPI installed.
+	mkdir -p $(PROJECT_PATH)/coverage/gatewayapi-integration
+#	Check `ginkgo help run` for command line options. For example to filtering tests.
+	$(GINKGO) \
+		--coverpkg $(INTEGRATION_COVER_PKGS) \
+		--output-dir $(PROJECT_PATH)/coverage/gatewayapi-integration \
+		--coverprofile cover.out \
+		-tags integration \
+		./tests/gatewayapi/...
+
+.PHONY: test-istio-env-integration
+test-istio-env-integration: clean-cov generate fmt vet ginkgo ## Requires kubernetes cluster with GatewayAPI and Istio installed.
+	mkdir -p $(PROJECT_PATH)/coverage/istio-integration
+#	Check `ginkgo help run` for command line options. For example to filtering tests.
+	$(GINKGO) \
+		--coverpkg $(INTEGRATION_COVER_PKGS) \
+		--output-dir $(PROJECT_PATH)/coverage/istio-integration \
+		--coverprofile cover.out \
+		-tags integration \
+		tests/istio/...
+
+.PHONY: test-integration
+test-integration: clean-cov generate fmt vet ginkgo ## Requires kubernetes cluster with at least one GatewayAPI provider installed.
+	mkdir -p $(PROJECT_PATH)/coverage/integration
+#	Check `ginkgo help run` for command line options. For example to filtering tests.
+	$(GINKGO) \
+		--coverpkg $(INTEGRATION_COVER_PKGS) \
+		--output-dir $(PROJECT_PATH)/coverage/integration \
+		--coverprofile cover.out \
+		-tags integration \
+		./controllers/...
```

Can we please keep `$(INTEGRATION_TEST_SUITE_PATHS)`? I use that a lot to filter the tests to run locally.

---

It can also be a more generic `INTEGRATION_TESTS_EXTRA_ARGS` that defaults to empty, if you prefer.


```suggestion
		$(INTEGRATION_TESTS_EXTRA_ARGS) ./controllers/...
```

I'd run it with, e.g.:

```sh
make test-integration INTEGRATION_TESTS_EXTRA_ARGS="--focus-file ./controllers/ratelimitpolicy_controller_test.go"
```

---

you know you can do that at Ginkgo level, right?

https://onsi.github.io/ginkgo/#focused-specs

I use `focused` feature a lot for doing exactly that.

---

Sure, but in that case I have to write the entire command myself, i.e. plus tags, progress, verbosity, etc, whereas this used be a handy extension on top of the standard way to run the integration tests.

---

added